app/components/Modals/TokenModal/TokenModal.jsx

```diff
@@ -90,28 +92,48 @@ class TokenModal extends Component<Props, State> {
     })
   }
 
+  shouldDisableSaveButton = () => {
+    const { tokens } = this.state
+
+    const invalidUserGeneratedTokens = tokens
+      .filter((token: TokenItemType) => token.isUserGenerated)
+      .filter((token: TokenItemType) => !token.scriptHash.length)
```

Comment:
you can use just one filter for both conditions - 
```
.filter((token: TokenItemType) => token.isUserGenerated && !token.scriptHash.length)
```

Reply:
👍